contrib/firebaseio/src/main/java/contrib/LoggingDoFn.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not  use this file except  in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package contrib;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * TODO: Insert description here. (generated by elibixby)
+ */
+public class LoggingDoFn<T> extends DoFn<T, T> {
+
+  private final String className;
+  private transient Logger logger;
+
+  public LoggingDoFn(Class<?> clazz){
+    className = clazz.getName();
+  }
+
+  @Override
+  public void startBundle(Context c) throws Exception {
+    logger = LoggerFactory.getLogger(Class.forName(className));
+  }
+
+  /* (non-Javadoc)
+   * @see DoFn#processElement(com.google.cloud.dataflow.sdk.transforms.DoFn.ProcessContext)
+   */
+  @Override
+  public void processElement(DoFn<T, T>.ProcessContext c) throws Exception {
+    logger.info(c.element().toString());
+    c.output(c.element());
+  }
+}
```

Comment:
Note that I specified a `PTransform`. Here's my carelessly-typed-in-here pseudocode, indented poorly to keep nesting straight:

```java
public class <T> LogElements extends PTransform<PCollection<T>, PDone> {
    private final Level level;
    public LogElements(Level level) { this.level = level; }
    private static class LoggingFn extends DoFn<T, Void> {
        private final Level level;
        public LoggingFn(Level level) { this.level = level; }     
        @Override void processElement(Context c) { LOG.log(level, "{}", c.element()); }
    }
    @Override PDone apply(PCollection<T> input) {
        input.apply(new LoggingFn(level));
        return PDone.in(input.getPipeline());
    }
}
```

Note that I deliberately put the `level` in the inner `DoFn` so that `LoggingFn` can be serialized without needing a copy of the outer `PTransform` as well.